plugins/by-name/neorg/default.nix

```
@@ -1,173 +1,222 @@
+          }
+          {
+            name = "info";
+            hl = "None";
+            level.__raw = "vim.log.levels.INFO";
+          }
+          {
+            name = "warn";
+            hl = "WarningMsg";
+            level.__raw = "vim.log.levels.WARN";
+          }
+          {
+            name = "error";
+            hl = "ErrorMsg";
+            level.__raw = "vim.log.levels.ERROR";
+          }
+          {
+            name = "fatal";
+            hl = "ErrorMsg";
+            level = 5;
+          }
+        ];
+      };
+
+      float_precision = defaultNullOpts.mkNullable types.float 0.01 ''
+        Can limit the number of decimals displayed for floats.
+      '';
+    };
+  };
+
+  settingsExample = {
+    load = {
+      "core.defaults".__empty = null;
+      "core.concealer".config = {
+        icon_preset = "varied";
+      };
+      "core.dirman".config = {
+        workspaces = {
+          work = "~/notes/work";
+          home = "~/notes/home";
+        };
       };
     };
   };
 
-  config =
-    let
-      setupOptions =
-        with cfg;
-        {
-          lazy_loading = lazyLoading;
-
-          logger = with logger; {
-            inherit plugin;
-            use_console = useConsole;
-            inherit highlights;
-            use_file = useFile;
-            inherit level;
-
-            modes = filter (v: v != null) (
-              mapAttrsToList (
-                mode: modeConfig:
-                helpers.ifNonNull' modeConfig {
-                  name = mode;
-                  inherit (modeConfig) hl level;
-                }
-              ) modes
-            );
-            float_precision = floatPrecision;
-          };
+  extraConfig = cfg: {
 
-          load = modules;
-        }
-        // cfg.extraOptions;
-
-      telescopeSupport = hasAttr "core.integrations.telescope" cfg.modules;
-    in
-    mkIf cfg.enable {
-      warnings =
-        (optional (telescopeSupport && (!config.plugins.telescope.enable)) ''
-          Telescope support for neorg (`core.integrations.telescope`) is enabled but the
-          telescope plugin is not.
-        '')
-        ++ (optional (telescopeSupport && (cfg.neorgTelescopePackage == null)) ''
-          Telescope support for neorg (`core.integrations.telescope`) is enabled but the
-          `neorgTelescopePackage` package is not set.
-        '')
-        ++ (optional ((hasAttr "core.defaults" cfg.modules) && (!config.plugins.treesitter.enable)) ''
-          Neorg's `core.defaults` module is enabled but `plugins.treesitter` is not.
-          Treesitter is required when using the `core.defaults`.
-        '');
-
-      extraPlugins = [
-        cfg.package
-      ] ++ (optional (telescopeSupport && cfg.neorgTelescopePackage != null) cfg.neorgTelescopePackage);
-
-      extraConfigLua = ''
-        require('neorg').setup(${helpers.toLuaObject setupOptions})
-      '';
-    };
+    extraPlugins = [ cfg.telescopeIntegration.package ];
```

```suggestion
    extraPlugins = lib.mkIf cfg.telescopeIntegration.enable [ cfg.telescopeIntegration.package ];
```

If we keep the enable option, we should check that they even want it installed.